### ISSUE TITLE
Refactor and fix Priority select visuals and aria labels

### ISSUE THREAD

**Issue by HenrikHartiala** [HSLdevcom/jore4#1442]
Part of accessibility audit findings
Refactor the visuals for priority select and also add / connect labels

We have two components, one for radio button and one for checkbox. Both should be changed.
PriorityCondition  is the checkbox version and PriorityForm is the radio button version.

Here are the designs for radio & checkbox and focus.
https://www.figma.com/file/ImSTkCqQn0nhVUtMcUm41P/Timetables-%7C-Routes-and-lines-%7C-JORE-4.0-UX?type=design&node-id=10478-333234&mode=design&t=OfcJluh047WDK2Rc-4

![Image](https://github.com/HSLdevcom/jore4/assets/82645403/587de492-ab87-4e7e-ae2a-96cc0a21af35)

 - [x] #1691
 - [x] #1692
 - [x] #1693
